File: fibonacci_memoization.c
#include<stdio.h>
long int table[100]= {0};
long int Fibonacci(int n){
    if(n==1 || n==2){
        return 1;
    }
    else{
        if(table[n]==0){
            table[n]=Fibonacci(n-1)+Fibonacci(n-2);
        }
        return table[n];
    }
}
int main(){
    int n;
    printf("Enter the value of n:\n");
    scanf("%d",&n);;
    printf("The %dth Fibonacci number is %ld\n",n,Fibonacci(n));
    return 0;
}

File: fibonacci_memoization.py
table = [0]*100
def Fibonacci(n):
    if(n==1 or n==2):
        return 1
    else:
        if(table[n]==0):
            table[n]=Fibonacci(n-1)+Fibonacci(n-2)
        return table[n]
n= int(input("Enter the value of n: "))
print(f"The {n}th Fibonacci number is {Fibonacci(n)}")
